src/core/fetchRequest.js

```diff
@@ -102,14 +108,26 @@ const requestFactory = (url, options) => {
             // create error
             let err;
             if (response.errorCode) {
-                err = new Error(
-                    `${response.errorCode} : ${response.errorMsg || response.errorMessage || response.error}`
+                err = new ApiError(
+                    `${response.errorCode} : ${response.errorMsg || response.errorMessage || response.error}`,
+                    response.errorCode,
+                    originalResponse
                 );
             } else {
-                err = new Error(`${responseCode} : Request error`);
+                err = new NetworkError(
+                    `${responseCode} : Request error`,
+                    responseCode || 0,
+                    originalResponse
+                );
             }
-            err.response = originalResponse;
             throw err;
+        })
+        .catch(err => {
+            if(!err.type){
+                //offline, CORS, etc.
+                return Promise.reject(new NetworkError(err.message, 0));
```

Comment:
Thought: what does `errorCode: 0` mean?

Reply:
https://fetch.spec.whatwg.org/#concept-network-error 0 means a network error: the request has not reached the server for various reason (no internet, no route, not allowed by CORS, etc.)